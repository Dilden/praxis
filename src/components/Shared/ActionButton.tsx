import {
  ButtonProps,
  createStyles,
  makeStyles,
  Button,
} from "@material-ui/core";

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      textTransform: "none",
    },
  })
);

const ActionButton = (props: ButtonProps) => {
  const classes = useStyles();

  return (
    <Button color="primary" classes={classes} {...props}>
      {props.children}
    </Button>
  );
};

export default ActionButton;
