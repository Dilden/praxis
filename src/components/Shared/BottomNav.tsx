import { useEffect, useState } from "react";
import Link from "next/link";
import Router, { useRouter } from "next/router";
import { BottomNavigation, BottomNavigationAction } from "@material-ui/core";
import { Person, Home, Group, Menu } from "@material-ui/icons";

import { NavigationPaths, ResourcePaths } from "../../constants/common";
import { navOpenVar } from "../../apollo/client/localState";
import { useCurrentUser } from "../../hooks";
import Messages from "../../utils/messages";
import { useReactiveVar } from "@apollo/client";

interface NavLinkProps {
  href: string;
  text?: string;
  icon?: React.ReactChild;
}

const NavLink = ({ href, icon, text }: NavLinkProps) => {
  return (
    <Link href={href}>
      <a style={{ color: "inherit" }}>{icon ? icon : text}</a>
    </Link>
  );
};

const BottomNav = () => {
  const currentUser = useCurrentUser();
  const [value, setValue] = useState<number>(0);
  const navDrawerOpen = useReactiveVar(navOpenVar);
  const userProfilePath = `${ResourcePaths.User}${currentUser?.name}`;
  const { asPath: currentPath } = useRouter();

  useEffect(() => {
    if (!navDrawerOpen)
      switch (currentPath) {
        case NavigationPaths.Home:
          setValue(0);
          break;
        case getMatching(ResourcePaths.Group):
        case NavigationPaths.Groups:
          setValue(1);
          break;
        case userProfilePath:
          setValue(2);
          break;
        default:
          setValue(3);
      }
  }, [currentPath, navDrawerOpen]);

  const getMatching = (path: string): string => {
    const match = currentPath.match(path);
    if (match) return currentPath;
    return "";
  };

  return (
    <BottomNavigation
      value={value}
      onChange={(_event, newValue) => {
        setValue(newValue);
      }}
      showLabels
    >
      <BottomNavigationAction
        onClick={() => Router.push(NavigationPaths.Home)}
        icon={<NavLink href={NavigationPaths.Home} icon={<Home />} />}
        label={Messages.navigation.home()}
      />

      <BottomNavigationAction
        onClick={() => Router.push(NavigationPaths.Groups)}
        icon={<NavLink href={NavigationPaths.Groups} icon={<Group />} />}
        label={Messages.navigation.groups()}
      />

      <BottomNavigationAction
        disabled={!currentUser}
        onClick={() => Router.push(userProfilePath)}
        icon={<NavLink href={userProfilePath} icon={<Person />} />}
        label={Messages.navigation.profile()}
      />

      <BottomNavigationAction
        onClick={() => navOpenVar(true)}
        label={Messages.navigation.menu()}
        icon={<Menu />}
      />
    </BottomNavigation>
  );
};

export default BottomNav;
