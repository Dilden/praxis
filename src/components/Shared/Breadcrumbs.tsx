import { useReactiveVar } from "@apollo/client";
import { Breadcrumbs, makeStyles, Typography } from "@material-ui/core";
import Link from "next/link";

import { breadcrumbsVar } from "../../apollo/client/localState";

const gray = "rgb(190, 190, 190)";

const useStyles = makeStyles({
  root: {
    marginBottom: 18,
  },
  separator: {
    color: gray,
  },
});

const CommonBreadcrumbs = () => {
  const breadcrumbs = useReactiveVar(breadcrumbsVar);
  const classes = useStyles();

  return (
    <Breadcrumbs classes={{ root: classes.root, separator: classes.separator }}>
      {breadcrumbs.map(({ label, href }) => {
        if (href)
          return (
            <Link href={href}>
              <a>
                <Typography style={{ color: gray }} key={label}>
                  {label}
                </Typography>
              </a>
            </Link>
          );
        return (
          <Typography style={{ color: "white" }} key={label}>
            {label}
          </Typography>
        );
      })}
    </Breadcrumbs>
  );
};

export default CommonBreadcrumbs;
