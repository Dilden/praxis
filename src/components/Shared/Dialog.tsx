import { forwardRef } from "react";
import {
  Button,
  makeStyles,
  Slide,
  Dialog,
  AppBar,
  Toolbar,
  IconButton,
  Typography,
  Container,
} from "@material-ui/core";
import { Close } from "@material-ui/icons";
import { TransitionProps } from "@material-ui/core/transitions";
import { CircularProgress } from "@material-ui/core";

const useStyles = makeStyles({
  titleWithCaption: {
    flex: 1,
  },
  appBar: {
    position: "relative",
    backgroundColor: "rgb(30, 30, 30)",
    marginBottom: 40,
  },
});

const DialogTransition = forwardRef(function Transition(
  props: TransitionProps & { children?: React.ReactElement },
  ref: React.Ref<unknown>
) {
  return <Slide direction="up" ref={ref} {...props} />;
});

interface Props {
  title: string;
  subtext?: string;
  actionLabel?: string;
  closingAction?: (_: any) => any;
  open: boolean;
  setOpen: (open: boolean) => void;
  children: React.ReactChild;
  loading?: boolean;
}

const CommonDialog = ({
  title,
  subtext,
  open,
  setOpen,
  actionLabel,
  closingAction,
  children,
  loading,
}: Props) => {
  const classes = useStyles();

  const handleCloseDialog = () => {
    setOpen(false);
  };

  return (
    <Dialog
      open={open}
      TransitionComponent={DialogTransition}
      onClose={handleCloseDialog}
      PaperProps={{
        style: {
          backgroundColor: "rgb(50, 50, 50)",
        },
      }}
      fullScreen
    >
      <AppBar className={classes.appBar}>
        <Toolbar>
          <IconButton color="inherit" onClick={handleCloseDialog}>
            <Close />
          </IconButton>
          <div className={classes.titleWithCaption}>
            <Typography
              variant="h6"
              style={subtext ? { marginTop: 5, marginBottom: -10 } : {}}
            >
              {title}
            </Typography>
            <Typography
              variant="caption"
              style={{ color: "rgb(150, 150, 150)" }}
            >
              {subtext}
            </Typography>
          </div>
          {actionLabel && closingAction && !loading && (
            <Button color="inherit" onClick={closingAction}>
              {actionLabel}
            </Button>
          )}
          {loading && <CircularProgress style={{ color: "white" }} />}
        </Toolbar>
      </AppBar>
      <Container maxWidth="md">{children}</Container>
    </Dialog>
  );
};

export default CommonDialog;
