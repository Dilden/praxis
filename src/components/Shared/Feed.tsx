import { CircularProgress } from "@material-ui/core";
import Motion from "../Motions/Motion";
import Post from "../Posts/Post";

interface Props {
  feed: FeedItem[];
  deleteMotion: (id: string) => void;
  deletePost: (id: string) => void;
}

const List = ({ feed, deleteMotion, deletePost }: Props) => {
  if (feed)
    return (
      <div style={{ marginBottom: "200px" }}>
        {feed
          .sort((a, b) => parseInt(b.createdAt) - parseInt(a.createdAt))
          .map((item) => {
            return item.__typename === "Motion" ? (
              <Motion
                motion={item as Motion}
                deleteMotion={deleteMotion}
                key={`motion-${item.id}`}
              />
            ) : (
              <Post
                post={item as Post}
                deletePost={deletePost}
                key={`post-${item.id}`}
              />
            );
          })}
      </div>
    );

  return <CircularProgress style={{ color: "white" }} />;
};

export default List;
