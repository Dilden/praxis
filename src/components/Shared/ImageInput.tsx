import { ChangeEvent, useRef } from "react";
import { Image } from "@material-ui/icons";

import styles from "../../styles/Shared/Shared.module.scss";

interface Props {
  setImage?: (image: File) => void;
  setImages?: (images: File[]) => void;
  multiple?: boolean;
  refreshKey: string;
}

const ImageInput = ({ setImage, setImages, multiple, refreshKey }: Props) => {
  const imageInput = useRef<HTMLInputElement>(null);

  const setImageState = (files: File[]) => {
    if (multiple && setImages) setImages(files);
    else if (setImage) setImage(files[0]);
  };

  return (
    <>
      <input
        multiple={multiple}
        type="file"
        accept="image/*"
        ref={imageInput}
        key={refreshKey}
        onChange={(e: ChangeEvent<HTMLInputElement>) =>
          e.target.files && setImageState([...e.target.files])
        }
        className={styles.imageInput}
      />
      <Image
        className={styles.imageInputIcon}
        onClick={() => imageInput.current?.click()}
        fontSize="large"
      />
    </>
  );
};

export default ImageInput;
