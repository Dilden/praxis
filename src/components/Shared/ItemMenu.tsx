import { Edit, Delete, MoreVert } from "@material-ui/icons";
import { Menu, MenuItem, IconButton } from "@material-ui/core";
import Link from "next/link";

interface Props {
  name?: string;
  itemId: string;
  itemType: string;
  anchorEl: null | HTMLElement;
  setAnchorEl: (el: null | HTMLElement) => void;
  deleteItem: (id: string) => void;
  ownItem: () => boolean;
  children?: React.ReactNode;
}

const ItemMenu = ({
  name,
  itemId,
  itemType,
  anchorEl,
  setAnchorEl,
  deleteItem,
  ownItem,
  children,
}: Props) => {
  const handleMenuButtonClick = (
    event: React.MouseEvent<HTMLButtonElement>
  ) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  if (ownItem())
    return (
      <>
        <IconButton onClick={handleMenuButtonClick}>
          <MoreVert style={{ color: "white" }} />
        </IconButton>
        <Menu
          anchorEl={anchorEl}
          keepMounted
          open={Boolean(anchorEl)}
          onClose={handleClose}
          PaperProps={{
            style: {
              backgroundColor: "rgb(65, 65, 65)",
            },
          }}
          MenuListProps={{
            style: {
              color: "white",
            },
          }}
        >
          <MenuItem>
            <Link href={`/${itemType}s/${name ? name : itemId}/edit`}>
              <a style={{ width: "100%" }}>
                <Edit
                  fontSize="small"
                  style={{
                    marginRight: "5",
                    transform: "rotateY(180deg)",
                  }}
                />
                Edit
              </a>
            </Link>
          </MenuItem>
          <MenuItem
            onClick={() =>
              window.confirm(
                `Are you sure you want to delete this ${itemType}?`
              ) && deleteItem(itemId)
            }
          >
            <Delete fontSize="small" style={{ marginRight: "5" }} />
            Delete
          </MenuItem>

          {children}
        </Menu>
      </>
    );

  return <></>;
};

export default ItemMenu;
