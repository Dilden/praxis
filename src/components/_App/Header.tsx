import Link from "next/link";
import { useEffect, useState } from "react";
import Router, { useRouter } from "next/router";
import jwtDecode from "jwt-decode";
import { useMutation, useReactiveVar } from "@apollo/client";
import MenuIcon from "@material-ui/icons/Menu";

import Messages from "../../utils/messages";
import { setAuthToken } from "../../utils/auth";
import { LOGOUT_USER, SET_CURRENT_USER } from "../../apollo/client/mutations";

import classNames from "classnames/bind";
import {
  useCurrentUser,
  useHasPermissionGlobally,
  useWindowSize,
} from "../../hooks";
import { Roles } from "../../constants";
import styles from "../../styles/Common/Header.module.scss";
import { headerKeyVar } from "../../apollo/client/localState";

const cx = classNames.bind(styles);

const Header = () => {
  const router = useRouter();
  const windowSize = useWindowSize();
  const [open, setOpen] = useState(false);
  const [logoutUser] = useMutation(LOGOUT_USER);
  const [setCurrentUser] = useMutation(SET_CURRENT_USER);
  const currentUser = useCurrentUser();
  const refreshKey = useReactiveVar(headerKeyVar);
  const [canManageRoles] = useHasPermissionGlobally(
    Roles.Permissions.ManageRoles,
    refreshKey
  );
  const [canManageUsers] = useHasPermissionGlobally(
    Roles.Permissions.ManageUsers,
    refreshKey
  );

  useEffect(() => {
    if (localStorage.jwtToken) {
      setAuthToken(localStorage.jwtToken);
      const decoded: User = jwtDecode(localStorage.jwtToken);
      setCurrentUserMutate(decoded);
      const currentTime = Date.now() / 1000;
      if (decoded.exp < currentTime) {
        logoutUserMutate();
      }
    }
  }, []);

  useEffect(() => {
    setOpen(false);
  }, [router.asPath, windowSize]);

  const setCurrentUserMutate = async (user: User) => {
    await setCurrentUser({
      variables: {
        id: user.id,
        name: user.name,
        email: user.email,
      },
    });
  };

  const logoutUserMutate = async () => {
    await logoutUser();
    Router.push("/users/login");
  };

  return (
    <>
      <nav className={styles.navbar} key={refreshKey}>
        <div className={styles.navbarBrand}>
          {router.asPath === "/" ? (
            <span onClick={() => router.reload()} role="button" tabIndex={0}>
              {Messages.brand()}
            </span>
          ) : (
            <Link href="/" passHref>
              {Messages.brand()}
            </Link>
          )}
        </div>
        <div
          className={cx(styles.navbarItems, {
            navbarItemsShow: open,
          })}
        >
          {canManageUsers && (
            <div className={styles.navbarItem}>
              <Link href="/users" passHref>
                <a className={styles.navbarItemText}>
                  {Messages.navigation.users()}
                </a>
              </Link>
            </div>
          )}

          <div className={styles.navbarItem}>
            <Link href="/groups" passHref>
              <a className={styles.navbarItemText}>
                {Messages.navigation.groups()}
              </a>
            </Link>
          </div>

          {canManageRoles && (
            <div className={styles.navbarItem}>
              <Link href="/roles" passHref>
                <a className={styles.navbarItemText}>
                  {Messages.navigation.roles()}
                </a>
              </Link>
            </div>
          )}

          {currentUser ? (
            <>
              <div className={styles.navbarItem}>
                <Link href={`/users/${currentUser.name}`} passHref>
                  <a className={styles.navbarItemText}>{currentUser.name}</a>
                </Link>
              </div>
              <div className={styles.navbarItem}>
                <div
                  onClick={() =>
                    window.confirm(Messages.prompts.logOut()) &&
                    logoutUserMutate()
                  }
                  role="button"
                  tabIndex={0}
                >
                  <a className={styles.navbarItemText}>
                    {Messages.users.actions.logOut()}
                  </a>
                </div>
              </div>
            </>
          ) : (
            <>
              <div className={styles.navbarItem}>
                <Link href="/users/login" passHref>
                  <a className={styles.navbarItemText}>
                    {Messages.users.actions.logIn()}
                  </a>
                </Link>
              </div>
              <div className={styles.navbarItem}>
                <Link href="/users/signup" passHref>
                  <a className={styles.navbarItemText}>
                    {Messages.users.actions.signUp()}
                  </a>
                </Link>
              </div>
            </>
          )}
        </div>
      </nav>

      <MenuIcon className={styles.menuButton} />
      <div
        className={styles.menuButtonTouchTarget}
        onClick={() => setOpen(!open)}
        role="button"
        tabIndex={-1}
      ></div>
    </>
  );
};

export default Header;
