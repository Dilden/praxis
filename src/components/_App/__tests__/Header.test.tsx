import { render, RenderResult, waitFor } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { mocked } from "ts-jest/utils";

import Messages from "../../../utils/messages";
import { Common } from "../../../constants";
import Header from "../Header";
import {
  useCurrentUser,
  useHasPermissionGlobally,
  useWindowSize,
} from "../../../hooks";
import { redeemedInviteToken } from "../../../utils/clientIndex";
import { mockNextUseRouter } from "../../Shared/__mocks__";

jest.mock("../../../hooks", () => ({
  useCurrentUser: jest.fn(),
  useHasPermissionGlobally: jest.fn(),
  useWindowSize: jest.fn(),
}));

jest.mock("../../../utils/clientIndex", () => ({
  redeemedInviteToken: jest.fn(),
}));

const mockUserName = "mockUserName";
const mockCurrentUser: CurrentUser = {
  id: "1",
  name: mockUserName,
  email: "test@email.com",
  isAuthenticated: true,
  __typename: Common.TypeNames.CurrentUser,
};
const mockInviteToken = "mockInviteToken";

const renderHeader = async (): Promise<RenderResult> => {
  mockNextUseRouter("/");
  return render(
    <MockedProvider>
      <Header />
    </MockedProvider>
  );
};

describe("Header", () => {
  beforeAll(() => {
    mocked(useWindowSize).mockReturnValueOnce([]);
  });

  it("should render Header component with groups and log in when user is logged out", async () => {
    mocked(useCurrentUser).mockReturnValueOnce(undefined);
    mocked(useHasPermissionGlobally).mockReturnValue([false]);
    mocked(redeemedInviteToken).mockReturnValueOnce(null);

    const { getByText, container } = await renderHeader();
    const groups = getByText(Messages.navigation.groups());
    const logIn = getByText(Messages.users.actions.logIn());

    await waitFor(() => expect(groups).toBeInTheDocument());
    await waitFor(() => expect(logIn).toBeInTheDocument());

    expect(container).toMatchSnapshot();
  });

  it("should render Header component with groups, sign up, and log in when user is logged out and has been invited", async () => {
    mocked(useCurrentUser).mockReturnValueOnce(undefined);
    mocked(useHasPermissionGlobally).mockReturnValue([false]);
    mocked(redeemedInviteToken).mockReturnValueOnce(mockInviteToken);

    const { getByText, container } = await renderHeader();
    const groups = getByText(Messages.navigation.groups());
    const signUp = getByText(Messages.users.actions.signUp());
    const logIn = getByText(Messages.users.actions.logIn());

    await waitFor(() => expect(groups).toBeInTheDocument());
    await waitFor(() => expect(signUp).toBeInTheDocument());
    await waitFor(() => expect(logIn).toBeInTheDocument());

    expect(container).toMatchSnapshot();
  });

  it("should render Header component with groups, profile, and log out when user is logged in", async () => {
    mocked(useCurrentUser).mockReturnValueOnce(mockCurrentUser);
    mocked(useHasPermissionGlobally).mockReturnValue([false]);

    const { getByText, container } = await renderHeader();
    const groups = getByText(Messages.navigation.groups());
    const logOut = getByText(Messages.users.actions.logOut());
    const profile = getByText(mockUserName);

    await waitFor(() => expect(groups).toBeInTheDocument());
    await waitFor(() => expect(profile).toBeInTheDocument());
    await waitFor(() => expect(logOut).toBeInTheDocument());

    expect(container).toMatchSnapshot();
  });

  it("should render Header component with groups, roles, users, and invites if user has permission", async () => {
    mocked(useHasPermissionGlobally).mockReturnValue([true]);

    const { getByText, container } = await renderHeader();
    const groups = getByText(Messages.navigation.groups());
    const roles = getByText(Messages.navigation.roles());
    const users = getByText(Messages.navigation.users());
    const invites = getByText(Messages.invites.labels.invites());

    await waitFor(() => expect(groups).toBeInTheDocument());
    await waitFor(() => expect(roles).toBeInTheDocument());
    await waitFor(() => expect(users).toBeInTheDocument());
    await waitFor(() => expect(invites).toBeInTheDocument());

    expect(container).toMatchSnapshot();
  });
});
