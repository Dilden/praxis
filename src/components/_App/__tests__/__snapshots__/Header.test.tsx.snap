// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header component with groups and log in when user is logged out 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="navbarBrand"
    >
      <a
        href="/"
      >
        praxis
      </a>
    </div>
    <div
      class="navbarItems"
    >
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/groups"
        >
          Groups
        </a>
      </div>
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/users/login"
        >
          Log in
        </a>
      </div>
    </div>
  </nav>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root menuButton"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
  <div
    class="menuButtonTouchTarget"
    role="button"
    tabindex="-1"
  />
</div>
`;

exports[`Header should render Header component with groups, profile, and log out when user is logged in 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="navbarBrand"
    >
      <a
        href="/"
      >
        praxis
      </a>
    </div>
    <div
      class="navbarItems"
    >
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/groups"
        >
          Groups
        </a>
      </div>
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/users/mockUserName"
        >
          mockUserName
        </a>
      </div>
      <div
        class="navbarItem"
      >
        <div
          role="button"
          tabindex="0"
        >
          <a
            class="navbarItemText"
          >
            Log out
          </a>
        </div>
      </div>
    </div>
  </nav>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root menuButton"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
  <div
    class="menuButtonTouchTarget"
    role="button"
    tabindex="-1"
  />
</div>
`;

exports[`Header should render Header component with groups, roles, users, and invites if user has permission 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="navbarBrand"
    >
      <a
        href="/"
      >
        praxis
      </a>
    </div>
    <div
      class="navbarItems"
    >
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/users"
        >
          Users
        </a>
      </div>
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/groups"
        >
          Groups
        </a>
      </div>
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/roles"
        >
          Roles
        </a>
      </div>
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/invites"
        >
          Invites
        </a>
      </div>
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/users/login"
        >
          Log in
        </a>
      </div>
    </div>
  </nav>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root menuButton"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
  <div
    class="menuButtonTouchTarget"
    role="button"
    tabindex="-1"
  />
</div>
`;

exports[`Header should render Header component with groups, sign up, and log in when user is logged out and has been invited 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="navbarBrand"
    >
      <a
        href="/"
      >
        praxis
      </a>
    </div>
    <div
      class="navbarItems"
    >
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/groups"
        >
          Groups
        </a>
      </div>
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/users/login"
        >
          Log in
        </a>
      </div>
      <div
        class="navbarItem"
      >
        <a
          class="navbarItemText"
          href="/users/signup"
        >
          Sign up
        </a>
      </div>
    </div>
  </nav>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root menuButton"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
  <div
    class="menuButtonTouchTarget"
    role="button"
    tabindex="-1"
  />
</div>
`;
